src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -746,11 +756,17 @@ protected void handleVersionUpgrades(Connection conn) throws SQLException, Inval
 	}
 
 	private String getPositionSHA() {
-		return getSchemaPositionSHA(serverID, position.getFile(), position.getOffset());
+		return getSchemaPositionSHA(serverID, position);
 	}
 
-	public static String getSchemaPositionSHA(Long serverID, String binlogFilename, Long binlogPosition) {
-		String shaString = String.format("%d/%s/%d", serverID, binlogFilename, binlogPosition);
+	public static String getSchemaPositionSHA(Long serverID, Position position) {
+		BinlogPosition binlogPosition = position.getBinlogPosition();
+		String shaString = String.format("%d/%s/%d/%d",
+			serverID,
+			binlogPosition.getFile(),
+			binlogPosition.getOffset(),
+			position.getLastHeartbeatRead()
+		);
```

Comment:
Yup, explained in the PR comment - I'm pretty sure that if this has an effect, it'll be short-lived and will fix itself once all clients are running the new version 🤞